src/manifest.ts

```diff
@@ -1538,6 +1560,61 @@ export class Manifest {
     }
     return this._pathsByComponent;
   }
+
+  /**
+   * Only return the auto-merge option if the release PR match filters. If no filter provided, do not auto merge. If
+   * multiple filters are provided we take the intersection (AND operation).
+   */
+  pullRequestAutoMergeOption(
+    pullRequest: ReleasePullRequest
+  ): AutoMergeOption | undefined {
+    const {versionBumpFilter, conventionalCommitFilter} = this.autoMerge || {};
+
+    // if the version bump do not match any provided filter value, do not auto-merge
+    const applyVersionBumpFilter = () => {
+      const versionBump =
+        pullRequest.version &&
+        pullRequest.previousVersion &&
+        pullRequest.version.compareBump(pullRequest.previousVersion);
+      return (
+        versionBump && versionBumpFilter?.find(filter => versionBump === filter)
+      );
+    };
+
+    // given two sets of type:scope items, if any item from commitSet isn't in filterSet do not auto-merge
+    const applyConventionalCommitFilter = () => {
+      if (pullRequest.conventionalCommits.length === 0) {
+        return false;
+      }
+      const commitSet = new Set(
+        pullRequest.conventionalCommits.map(
+          commit => `${commit.type}:${commit.scope}`
+        )
+      );
+      const filterSet = new Set(
+        conventionalCommitFilter!.map(
+          filter => `${filter.type}:${filter.scope}`
+        )
+      );
+      for (const n of commitSet) {
+        if (!filterSet.has(n)) {
+          return false;
+        }
+      }
+      return true;
+    };
+
+    const selected =
+      conventionalCommitFilter?.length && conventionalCommitFilter.length
+        ? applyConventionalCommitFilter() && applyVersionBumpFilter()
+        : conventionalCommitFilter?.length
+        ? applyConventionalCommitFilter()
+        : versionBumpFilter?.length
+        ? applyVersionBumpFilter()
+        : // no filter provided
+          false;
+    return selected ? this.autoMerge : undefined;
+  }
 }
```

Comment:
@rattrayalex @RobertCraigie This method makes it possible to filter based on conventional commits type and scopes, and version bumps. You can see the test case I added to `test/manifest.ts` to see it in action.

Reply:
Good point, I only considered situations with both `type` and `scope` because that's what seemed to have been agreed upon with Robert (only auto-merge for `fix(api)`). But it would make sense for the scope to be optional. That shouldn't be a big change.

Reply:
err, we agreed to auto-merge for `feat(api)` or _any_ `fix`

Reply:
Then I missed that part :)
Reading your comment on the Linear I understood it was only for `feat(api)`, but that's not a big deal, we can easily add support for all `fix` commits 👍.

Reply:
sweet! thanks sam! sorry for the lack of clarity, appreciate your PR comments to highlight the important bits!!

Reply:
And... done, with tests covering exactly your use case.